File: compiler/bench/dupn.fs
\ expected: 300000000
\ Dup at various depths, 300M times
: main
  0 1 2 3                   ( a b c d )
  300000000 0 do
    dup drop                ( a b c d ) dup top
    over nip                ( a b c d ) dup second via over
    >r dup r> swap drop     ( a b c d ) more stack work
    >r >r >r 1 + r> r> r>   ( a' b c d )
  loop
  drop drop drop . cr
;

File: compiler/bench/dupn.c
// expected: 300000000
// Dup at various depths, 300M times
#include <stdio.h>

int main(void) {
    long a = 0, b = 1, c = 2, d = 3;
    for (long i = 0; i < 300000000L; i++) {
        long tmp = d;  // dup top
        (void)tmp;
        tmp = c;       // dup second
        (void)tmp;
        a = a + 1;
    }
    (void)b;
    (void)c;
    (void)d;
    printf("%ld\n", a);
    return 0;
}
